web/client/components/tutorial/Tutorial.jsx

```diff
@@ -0,0 +1,190 @@
+        intro: React.PropTypes.bool,
+        introPosition: React.PropTypes.number,
+        rawSteps: React.PropTypes.array,
+        nextLabel: React.PropTypes.string,
+        showCheckbox: React.PropTypes.bool,
+        defaultStep: React.PropTypes.object,
+        error: React.PropTypes.object,
+
+        steps: React.PropTypes.array,
+        stepIndex: React.PropTypes.number,
+        run: React.PropTypes.bool,
+        autoStart: React.PropTypes.bool,
+        keyboardNavigation: React.PropTypes.bool,
+        resizeDebounce: React.PropTypes.bool,
+        resizeDebounceDelay: React.PropTypes.number,
+        holePadding: React.PropTypes.number,
+        scrollOffset: React.PropTypes.number,
+        scrollToSteps: React.PropTypes.bool,
+        scrollToFirstStep: React.PropTypes.bool,
+        showBackButton: React.PropTypes.bool,
+        showOverlay: React.PropTypes.bool,
+        allowClicksThruHole: React.PropTypes.bool,
+        showSkipButton: React.PropTypes.bool,
+        showStepsProgress: React.PropTypes.bool,
+        tooltipOffset: React.PropTypes.number,
+        type: React.PropTypes.string,
+        disableOverlay: React.PropTypes.bool,
+        debug: React.PropTypes.bool,
+
+        actions: React.PropTypes.object
+    },
+    getDefaultProps() {
+        return {
+            toggle: false,
+            preset: 'map',
+            presetList: {},
+            intro: true,
+            introPosition: (window.innerHeight - 348) / 2,
+            rawSteps: [],
+            nextLabel: 'next',
+            showCheckbox: true,
+            defaultStep: {
+                title: '',
+                text: '',
+                position: 'bottom',
+                type: 'click',
+                allowClicksThruHole: true
+            },
+            error: {
+                style: errorStyle,
+                text: <I18N.Message msgId="tutorial.error"/>
+            },
+
+            steps: [],
+            stepIndex: 0,
+            run: true,
+            autoStart: true,
+            keyboardNavigation: true,
+            resizeDebounce: false,
+            resizeDebounceDelay: 200,
+            holePadding: 0,
+            scrollOffset: 20,
+            scrollToSteps: true,
+            scrollToFirstStep: false,
+            showBackButton: true,
+            showOverlay: true,
+            allowClicksThruHole: false,
+            showSkipButton: false,
+            showStepsProgress: true,
+            tooltipOffset: 10,
+            type: 'continuous',
+            disableOverlay: false,
+            debug: false,
+
+            actions: {
+                onSetup: () => {},
+                onStart: () => {},
+                onUpdate: () => {},
+                onToggle: () => {},
+                onReset: () => {}
+            }
+        };
+    },
+    componentWillMount() {
+        let rawSteps = this.props.rawSteps.length > 0 ? this.props.rawSteps : this.props.presetList[this.props.preset] || [];
```

Comment:
We try to not use lifecycle methods, if not necessary. This is state initialization code, I presume so should be developed as actions / reducers code